### PR TITLE
Update observability AI assistant docs with latest changes

### LINKED ISSUE #3652: [Request]: Update AI assistant documentation for 8.13
### Description

Request submitted by @grabowskit 
~We need to get the `query` (previously called `esql`) and `context` functions into the 8.13 documentation for the AI Assistant.~ It sounds like we do not really want to externalize these functions after all based on Dario's feedback on the draft PR.
AWS Bedrock support is coming too.

### Resources

https://www.elastic.co/blog/whats-new-elastic-observability-8-13-0

@grabowskit Can you point me to related issues/PRs that describe the new functions? Thanks!

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

TBD

@dgieselaar Can you identify any differences for stateful vs serverless?

### What release is this request related to?

8.13

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @dgieselaar 

**Stakeholders:** @tomgab

### PR DESCRIPTION
**Pull request by dedemorton**
Adds changes to stateful for #3652. Note that we decided not to document "internal" functions.

Also alphabetizes the table of functions. 

Preview link: https://observability-docs_bk_3663.docs-preview.app.elstc.co/guide/en/observability/master/obs-ai-assistant.html

Question: Why don't we alphabetize the list of functions in the UI? I know we sometimes like to put the most commonly used options first, but in my experience that kind of organizational approach falls apart over time.

TODO (after merging):

- [ ] Port changes to serverless docs.

## Open questions (these questions also appear inline in the diff):
- [x] What is the best link to use for AWS Bedrock security? The [one I’ve provided in the updated docs](https://docs.aws.amazon.com/bedrock/latest/userguide/security-iam.html), or the link that’s provided in the [AI Assistant](https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html)? _RESOLVED: Leaving the link as-is._
- [x] Should we replace the table of functions in the docs with the UI instructions for showing the list of functions (like we did in serverless)? There are some good arguments for doing this, but I wonder if seeing this info might actually be useful for people who are exploring the docs to figure out what is possible. If so, it’s better to keep the table, IMO. In not, then maybe just telling users how to see the list in the UI is good enough. _RESOLVED: Removing the docs for internal functions and leaving the other functions documented._
- [x] Has the "recall" function really has been renamed to "context". Can we just remove the description or do we need to mention that it’s been renamed? _RESOVLED: Removing the docs for these functions._
- [x] Is the list of functions documented in this PR correct for 8.13? Only a subset are actually shown in the UI. None of the apm functions are shown in the UI right now (maybe because I don't have APM set up?):
image
  If I ask the AI Assistant what functions are available, it'll give me a long list, including context and query, but that list does not show up in the UI when I click **Select a function**. _RESOLVED: We will remove the internal functions._
- [x] Is the description of the query function  in this PR true? (Or is AI making stuff up?) _RESOVLED: Yes, but we will remove the query function because it's an LLM-only function._

### CODE REVIEW

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 1]
```diff
@@ -47,17 +53,24 @@ Elastic does not control third-party tools, and assumes no responsibility or lia
...
-. From *{stack-manage-app}* -> *{connectors-ui}* in {kib}, create a {kibana-ref}/openai-action-type.html[OpenAI connector].
+. From *{stack-manage-app}* -> *{connectors-ui}* in {kib}, create an {kibana-ref}/openai-action-type.html[OpenAI] or {kibana-ref}/bedrock-action-type.html[Amazon Bedrock] connector.
 . Authenticate communication between {observability} and the AI provider by providing the following information:
-.. Enter the AI provider's API endpoint URL in the *URL* field.
-.. Enter the API key you created in the previous step in the *API Key* field.
+.. In the *URL* field, enter the AI provider's API endpoint URL.
+.. Under *Authentication*, enter the API key or access key/secret you created in the previous step.
+
+//TODO: Resolve this question:
+**REVIEWERS:** What is the best link to use for AWS Bedrock security? The one I've provided above, or the link that's provided in the product? (https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html)
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:73]: I think the Bedrock one is more specific, but I barely work with AWS :(

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 2]
```diff
@@ -179,19 +192,37 @@ The AI Assistant uses functions to include relevant context in the chat conversa
 
 The following table lists available functions:
 
+//TODO: Resolve this question:
+**REVIEWERS**: Is the list of functions here correct? Only a subset are actually shown in the UI. None of the apm functions are shown in the UI right now.
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:196]: Yes, some functions are hidden from the user as they're only available to the Assistant, and APM functions only show up when there is APM data in the cluster.

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 3]
```diff
@@ -179,19 +192,37 @@ The AI Assistant uses functions to include relevant context in the chat conversa
...
+
+//TODO: Resolve this question:
+**REVIEWERS**: Should we replace this table with the UI instructions for showing the list of functions (like we did in serverless)?
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:199]: I'm fine with leaving them here, but we should consider them an implementation detail while they're still in beta. They can easily change or be removed entirely (like we did with the lens function).

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 4]
```diff
@@ -179,19 +192,37 @@ The AI Assistant uses functions to include relevant context in the chat conversa
...
+`get_dataset_info`:: Get information about available indices and datasets, and the fields available on them.
+`kibana`:: Call {kib} APIs on your behalf.
+`lens`:: Create custom visualizations, using {kibana-ref}/lens.html[Lens], that you can add to dashboards.
+`recall`:: Recall previous learning. This function has been renamed to `context`.
+`summarize`:: Summarize parts of the conversation.
+`query`:: Generate, execute, or visualize a query based on the user's request. This function can also explain how ES/QL works and convert queries from one language to another.
+
+//TODO: Resolve this question:
+**REVIEWERS**: Has the "recall" function really has been renamed to "context". Can we just remove the description or do we need to mention that it's been renamed here?
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:222]: Yes, `recall` has been renamed to `context`. Let's remove `recall`, `context` and `lens` at least.

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 5]
```diff
@@ -179,19 +192,37 @@ The AI Assistant uses functions to include relevant context in the chat conversa
...
+
+//TODO: Resolve this question:
+**REVIEWERS**: Is the above description of the query function true or is the AI making stuff up? My original description said something like: "Construct and render an ES/QL query using the LLM in a multi-step process to improve the output." Also do the functions I've added here actually exist, or was AI making that up too?
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:225]: Yes this is true, but I would also remove `query` - it's really an LLM-only function.

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 6]
```diff
@@ -177,21 +187,30 @@ beta::[]
...
 `get_apm_service_summary`:: Get a summary of a single service, including the language, service version, deployments, the environments, and the infrastructure that it is running in. For example, the number of pods and a list of their downstream dependencies. It also returns active alerts and anomalies.
 `get_apm_services_list`:: Get the list of monitored services, their health statuses, and alerts.
+`get_apm_timeseries`:: Display different APM metrics (such as throughput, failure rate, or latency) for any service or all services and any or all of their dependencies. Displayed both as a time series and as a single statistic. Additionally, the function  returns any changes, such as spikes, step and trend changes, or dips. You can also use it to compare data by requesting two different time ranges, or, for example, two different service versions.
+
+TIP: Internal functions used by the AI Assistant are not shown here because they are not available for you to suggest.
+You can learn more about these functions by asking the AI Assistant about them. For example, you can ask the AI Assistant,
+"What is the context function?"
+
+//TODO: Make sure execute_query is not an internal function.
```
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:213]:
```suggestion
```

**Comment by dedemorton** [comment]
@dgieselaar This PR is ready for a final review.

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 7]
```diff
@@ -33,6 +38,7 @@ The AI assistant requires the following:
 * An account with a third-party generative AI provider that supports function calling. The Observability AI Assistant supports the following providers:
 ** OpenAI `gpt-4`+.
 ** Azure OpenAI Service `gpt-4`(0613) or `gpt-4-32k`(0613) with API version `2023-07-01-preview` or more recent.
+** AWS Bedrock, specifically the Anthropic Claude models.
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:41]: Right now we do not support Claude 3: https://github.com/elastic/security-team/issues/8857

File: docs/en/observability/observability-ai-assistant.asciidoc [hunk 8]
```diff
@@ -177,21 +187,29 @@ beta::[]
...
 `get_apm_service_summary`:: Get a summary of a single service, including the language, service version, deployments, the environments, and the infrastructure that it is running in. For example, the number of pods and a list of their downstream dependencies. It also returns active alerts and anomalies.
 `get_apm_services_list`:: Get the list of monitored services, their health statuses, and alerts.
+`get_apm_timeseries`:: Display different APM metrics (such as throughput, failure rate, or latency) for any service or all services and any or all of their dependencies. Displayed both as a time series and as a single statistic. Additionally, the function  returns any changes, such as spikes, step and trend changes, or dips. You can also use it to compare data by requesting two different time ranges, or, for example, two different service versions.
+
+TIP: Internal functions used by the AI Assistant are not shown here because they are not available for you to suggest.
+You can learn more about these functions by asking the AI Assistant about them. For example, you can ask the AI Assistant,
+"What is the context function?"
```
Comment (dgieselaar) [inline docs/en/observability/observability-ai-assistant.asciidoc:211]: I'm not sure if we should add this. We have four levels of visibility - available all the time, available to the user + Elastic, available to the LLM + Elastic, and Elastic only. E.g. we might move `context` to be an Elastic-only function. I would consider these an implementation detail where it is somewhat of a bug (but inherent to LLMs and not fixable by us) that the LLMs can "tell" the user about these.

**Review by dgieselaar — DISMISSED** [review]
Approving ahead of time, but specifically lack of support for Claude 3 should probably be mentioned

File: docs/en/observability/observability-ai-assistant.asciidoc (hunk 8 above)
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:211]:
```suggestion
```

**Review by dgieselaar — DISMISSED** [review]

File: docs/en/observability/observability-ai-assistant.asciidoc (hunk 7 above)
Comment (dedemorton) [inline docs/en/observability/observability-ai-assistant.asciidoc:41]:
```suggestion
** AWS Bedrock, specifically the Anthropic Claude models. Note that we do not currently support Claude 3.0 models.
```

Comment (dedemorton) [reply docs/en/observability/observability-ai-assistant.asciidoc]: Added as a restriction. We'll need to make sure the restriction gets removed when support is added.

File: docs/en/observability/observability-ai-assistant.asciidoc (hunk 8 above)
Comment (dedemorton) [reply docs/en/observability/observability-ai-assistant.asciidoc]: Removed the tip about internal functions.

**Review by mdbirnstiehl — APPROVED** [review]
LGTM!

**Pull request merged**